feed/obs/thread.js

```diff
@@ -32,19 +32,24 @@ exports.create = function (api) {
     })
 
     var backlinks = api.backlinks.obs.for(rootId)
+
+    // wrap computed in a map to turn into individual observables
     var replies = map(computed(backlinks, (msgs) => {
-      return msgs.filter(msg => {
+      return sort(msgs.filter(msg => {
         const { type, branch } = msg.value.content
         return type !== 'vote' && !isBlocked(msg) && (root(msg) === rootId || matchAny(branch, rootId))
-      })
+      }))
     }), x => Value(x), {
       // avoid refresh of entire list when items added
       comparer: (a, b) => a === b
     })
 
-    var messages = computed([prepend, replies], (prepend, replies) => {
-      return sort([...prepend, ...replies])
-    })
+    // append the root message to the sorted replies list
+    // -------------------------
+    // concat preserves the individual observable messages so that clients don't need to 
+    // rerender the entire list when an item is added (map will only be called for new items)
+    // (we can't use a computed here as it would squash the individual observables into a single one)
```

Comment:
hidden mutant features which are awesome but I didn't know about and lead me to writing stupid code 
T_T

Reply:
#somebodyshould copy and paste that comment into the docs....